mongo/single_result_test.go

```diff
@@ -49,3 +50,36 @@ func TestSingleResult(t *testing.T) {
 		assert.Equal(t, ErrNoDocuments, sr.Err(), "expected error %v, got %v", ErrNoDocuments, sr.Err())
 	})
 }
+
+func TestNewSingleResultFromBytes(t *testing.T) {
+	// Mimic a FindOne response from the server.
```

Comment:
```suggestion
	// Mock a document in SingleResult.
```

Nit: I interpret `response` to mean the raw server response to a cursor operation like `find` or `aggregate`. For example, running `db.runCommand({find: "foo", filter: {}})` in the shell shows the raw server response:

```
{
  cursor: {
    firstBatch: [
      { _id: ObjectId("61a7cbe3f6959812549d1134"), x: 1 },
      { _id: ObjectId("61a7ce23f6959812549d1135"), x: 2 }
    ],
    id: Long("0"),
    ns: 'test.foo'
  },
  ok: 1,
  '$clusterTime': {
    clusterTime: Timestamp({ t: 1638386659, i: 2 }),
    signature: {
      hash: Binary(Buffer.from("0000000000000000000000000000000000000000", "hex"), 0),
      keyId: Long("0")
    }
  },
  operationTime: Timestamp({ t: 1638386659, i: 2 })
}
```

That response is the expected input to `NewCursorResponse`. I suggest renaming `findOneResponse` to `findOneResult` and `findOneResponseBytes` to `findOneResultBytes`.

Reply:
Great point. I've tried to use the word `result` instead of `response`, as I agree with your distinction.